src/Microsoft.CorrelationVector/CorrelationVector.cs

```diff
@@ -1,397 +1,159 @@
-        public static CorrelationVector Extend(string correlationVector)
-        {
-            if (CorrelationVector.IsImmutable(correlationVector))
-            {
-                return CorrelationVector.Parse(correlationVector);
-            }
-
-            CorrelationVectorVersion version = CorrelationVector.InferVersion(
-                correlationVector, CorrelationVector.ValidateCorrelationVectorDuringCreation);
-
-            if (CorrelationVector.ValidateCorrelationVectorDuringCreation)
-            {
-                CorrelationVector.Validate(correlationVector, version);
-            }
+        /// Gets or sets a value indicating whether or not to validate the correlation
+        /// vector on creation.
+        /// </summary>
+        public static bool ValidateCorrelationVectorDuringCreation { get; set; }
 
-            if (CorrelationVector.IsOversized(correlationVector, 0, version))
-            {
-                return CorrelationVector.Parse(correlationVector + CorrelationVector.TerminationSign);
-            }
-
-            return new CorrelationVector(correlationVector, 0, version, false);
-        }
-
-        /// <summary>
-        /// Creates a new correlation vector by applying the Spin operator to an existing value.
-        /// This should be done at the entry point of an operation.
-        /// </summary>
-        /// <param name="correlationVector">
-        /// Taken from the message header indicated by <see cref="HeaderName"/>.
-        /// </param>
-        /// <returns>A new correlation vector extended from the current vector.</returns>
-        public static CorrelationVector Spin(string correlationVector)
-        {
-            SpinParameters defaultParameters = new SpinParameters
-            {
-                Interval = SpinCounterInterval.Coarse,
-                Periodicity = SpinCounterPeriodicity.Short,
-                Entropy = SpinEntropy.Two
-            };
-
-            return CorrelationVector.Spin(correlationVector, defaultParameters);
-        }
-
-        /// <summary>
-        /// Creates a new correlation vector by applying the Spin operator to an existing value.
-        /// This should be done at the entry point of an operation.
-        /// </summary>
-        /// <param name="correlationVector">
-        /// Taken from the message header indicated by <see cref="HeaderName"/>.
-        /// </param>
-        /// <param name="parameters">
-        /// The parameters to use when applying the Spin operator.
-        /// </param>
-        /// <returns>A new correlation vector extended from the current vector.</returns>
-        public static CorrelationVector Spin(string correlationVector, SpinParameters parameters)
+        /// <summary>
+        /// This is the maximum vector length before it has to be reset or terminated.
+        /// </summary>
+        internal const byte MaxVectorLength = 127;

```

Comment:
If this isn't being used, lets remove it

Reply:
This will be used in cV v3 and later.